mlrun/projects/project.py

```diff
@@ -1498,12 +1497,14 @@ def log_artifact(
         artifact_path = mlrun.utils.helpers.template_artifact_path(
             artifact_path, self.metadata.name
         )
-        producer = ArtifactProducer(
-            "project",
-            self.metadata.name,
-            self.metadata.name,
-            tag=self._get_hexsha() or str(uuid.uuid4()),
-        )
+        producer = self._resolve_artifact_producer(item)
+        if producer.name != self.metadata.name:
+            # the artifact producer is retained, so log it only if it doesn't already exist
+            if existing_artifact := self._resolve_existing_artifact(
```

Comment:
That's a good question actually. 
As per @theSaarco we shouldn't re-register artifacts if we retain their producer, but indeed the spec etc can change.

Reply:
Good question indeed.
Of course this is only expected if we're re-importing on the same env, since it's a run-id that already exists in the system. Even then, I tend to think it's a strange situation in which the same run-id had generated a different version of the same artifact. It seems more likely that the same artifact was modified after the run was done, in which case it would be probably better to maintain the manual changes done to the artifact.
I guess if the user wants to "reset" the artifact, it will need to be deleted before the import.

Reply:
ok, that makes sense. I suggest adding a log that says the artifact is already registered so we're skipping it.